flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/AbstractRestHandler.java

```diff
@@ -105,7 +106,8 @@ protected void respondToRequest(ChannelHandlerContext ctx, HttpRequest httpReque
 					messageHeaders.getResponseStatusCode(),
 					responseHeaders);
 			}
-		});
+		}).whenComplete((P resp, Throwable throwable) -> processingFinishedFuture.complete(null));
```

Comment:
I think we are swallowing potential exceptions here. I think it would be better to do something like
```
return response.whenComplete(...).thenApply(ignored -> null)
```

That way we would also get rid of the `processingFinishedFuture`.